### PR TITLE
Fixed Endless kick Loop in Velocity Bridge Implementation

### PR DESCRIPTION
**Pull request by davidmayr**
This pull request includes:

- [ ] breaking changes
- [x] no breaking changes

### Changes made to the repository:

I fixed the bug that I documented here: https://discord.com/channels/325362837184577536/536594898321670154/788033187770400819

For Users that aren't on the Discord/don't speak German:

If you connect to Velocity on a 1.16 network in an older version of Minecraft (1.8 for example) the console is spammed in CloudNet, Velocity and the target server. The server always kicks the player for a too old version and your Velocity plugin sends the player back to the specific server that kicked the player.... This causes an endless loop until disconnection.

What I have fixed:
I compared the Bungeecord and the Velocity implementations and implemented the missing lines in the Velocity implementation.

### Documentation of test results:

I tested with one and two Lobbys. Everything works fine.

With 2 Lobbys:
CloudNet seems to send the player twice to Lobby-1 and once to Lobby-2 and then: KICK. Works perfectly. No endless loop.

With 1 Lobby:
It again sends the player twice to Lobby-1 and then kicks him.

### Related issues/discussions:

https://discord.com/channels/325362837184577536/536594898321670154/788033187770400819

### CODE REVIEW

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java [hunk 1]
```diff
@@ -72,8 +72,14 @@ public void handle(ServerConnectedEvent event) {
 
     @Subscribe
     public void handle(KickedFromServerEvent event) {
-        VelocityCloudNetHelper.getNextFallback(event.getPlayer())
+        if(!event.getPlayer().isActive())
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java]: Please don't return here, just do 
```java
if (event.getPlayer().isActive()) {
  // code
}
```

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java [hunk 2]
```diff
@@ -72,8 +72,14 @@ public void handle(ServerConnectedEvent event) {
...
+            return;
+
+        BridgeProxyHelper.handleConnectionFailed(event.getPlayer().getUniqueId(), event.getServer().getServerInfo().getName());
+
+        VelocityCloudNetHelper.getNextFallback(event.getPlayer(), event.getServer())
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java]: As you called `handleConnectionFailed` there is no need of passing the server to the method, it simply generates an overload

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java [hunk 3]
```diff
@@ -72,8 +72,14 @@ public void handle(ServerConnectedEvent event) {
 
     @Subscribe
     public void handle(KickedFromServerEvent event) {
-        VelocityCloudNetHelper.getNextFallback(event.getPlayer())
+        if(!event.getPlayer().isActive())
+            return;
+
+        BridgeProxyHelper.handleConnectionFailed(event.getPlayer().getUniqueId(), event.getServer().getServerInfo().getName());
+
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java]: No need for the empty line

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java [hunk 4]
```diff
@@ -72,8 +72,14 @@ public void handle(ServerConnectedEvent event) {
...
+        VelocityCloudNetHelper.getNextFallback(event.getPlayer(), event.getServer())
                 .ifPresent(registeredServer -> event.setResult(KickedFromServerEvent.RedirectPlayer.create(registeredServer)));
+
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java]: Same here

**Review by derklaro — CHANGES_REQUESTED** [review]
LGTM but please take care of the following things:

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetHelper.java [hunk 5]
```diff
@@ -90,11 +90,10 @@ public static NetworkConnectionInfo createNetworkConnectionInfo(Player player) {
                 BridgeHelper.createOwnNetworkServiceInfo()
         );
     }
-
-    public static Optional<RegisteredServer> getNextFallback(Player player) {
+    public static Optional<RegisteredServer> getNextFallback(Player player, RegisteredServer registeredServer) {
         return BridgeProxyHelper.getNextFallback(
                 player.getUniqueId(),
-                player.getCurrentServer().map(ServerConnection::getServerInfo).map(ServerInfo::getName).orElse(null),
```
Comment (juliarn) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetHelper.java]: Everything in this class should be fine, why did you change this?

Comment (davidmayr) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetHelper.java]: This actually fixed it for a single Lobby. This was my first try.

I guess I will revert this then. But why is it in the Bungeecord implementation when it shouldn't be here?

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java (hunk 2 above)
Comment (davidmayr) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java]: (Copy and Paste from my Comment to @juliarn's review) 
This actually fixed it for a single Lobby. This was my first try.

I guess I will revert this then. But why is it in the Bungeecord implementation when it shouldn't be here?

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetHelper.java (hunk 5 above)
Comment (juliarn) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetHelper.java]: Because the velocity api returns optionals, which make a way cleaner api usage possible. Bungee is not doing this, so the code is a bit different.

Comment (davidmayr) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/VelocityCloudNetHelper.java]: Well I changed it because the current server was null but I think with handleConnectionFailed I can just ignore this. Reverting this now.

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java (hunk 3 above)
Comment (davidmayr) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java]: What empty line? The first or the second?

Since I did some changes now could you review them too? Is it fine now?

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java [hunk 6]
```diff
@@ -72,15 +72,13 @@ public void handle(ServerConnectedEvent event) {
 
     @Subscribe
     public void handle(KickedFromServerEvent event) {
-        if(!event.getPlayer().isActive())
-            return;
+        if(event.getPlayer().isActive()) {
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java]: Can you please reformat to `if (..`?

File: cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java (hunk 3 above)
Comment (derklaro) [reply cloudnet-modules/cloudnet-bridge/src/main/java/de/dytanic/cloudnet/ext/bridge/velocity/listener/VelocityPlayerListener.java]: Looks good now

**Review by derklaro — APPROVED** [review]
LGTM now

**Review by juliarn — APPROVED** [review]

**Comment by juliarn** [comment]
Thank you
